### PR TITLE
Splines simple

### PR DESCRIPTION
**Pull request by tpadioleau**

### CODE REVIEW

File: include/ddc/kernels/splines/matrix_sparse.hpp [hunk 1]
```diff
@@ -116,91 +90,39 @@ unsigned int default_preconditionner_max_block_size() noexcept
...
+private:
+    std::unique_ptr<gko::matrix::Dense<double>> m_matrix_dense;
 
-    Kokkos::View<double*, Kokkos::HostSpace> m_data;
+    std::shared_ptr<matrix_sparse_type> m_matrix_sparse;
 
-    std::unique_ptr<gko::solver::Bicgstab<double>::Factory> m_solver_factory;
+    std::shared_ptr<gko::solver::Bicgstab<double>> m_solver;
 
     int m_cols_per_par_chunk; // Maximum number of columns of B to be passed to a Ginkgo solver
```
Comment (blegouix) [inline include/ddc/kernels/splines/matrix_sparse.hpp:106]: Should I rename this as `m_chunk_size` everywhere ?

Comment (tpadioleau) [reply include/ddc/kernels/splines/matrix_sparse.hpp:106]: Yes good idea

File: benchmarks/splines.cpp [hunk 2]
```diff
@@ -179,57 +179,51 @@ static void characteristics_advection(benchmark::State& state)
...
+        ->Ranges({{64, 1024}, {100000, 100000}, {64,65535}, {preconditionner_max_block_size_ref, preconditionner_max_block_size_ref}})
         ->MinTime(3)->UseRealTime();
 */
 /*
 BENCHMARK(characteristics_advection)
         ->RangeMultiplier(2)
-        ->Ranges({{64, 1024}, {100000, 100000}, {cols_per_par_chunk_ref, cols_per_par_chunk_ref}, {1, 10000}, {preconditionner_max_block_size_ref, preconditionner_max_block_size_ref}})
+        ->Ranges({{64, 1024}, {100000, 100000}, {cols_per_chunk_ref, cols_per_chunk_ref}, {preconditionner_max_block_size_ref, preconditionner_max_block_size_ref}})
         ->MinTime(3)->UseRealTime();
 */
```
Comment (tpadioleau) [inline benchmarks/splines.cpp:214]: @blegouix can we remove this one ? seems redundant with the uncommented one

**Review by tpadioleau — COMMENTED** [review]
@blegouix I think we are good, what do you think ?

**Comment by blegouix** [comment]
Yes I think we are good to merge too

**Pull request merged**